TodoList/bot/teamsBot.ts

```diff
@@ -1,241 +1,301 @@
+                tabResp = await createSharedwithMeResponse(
+                    user,
+                    tokenResponse.token,
+                    this.checkFectchTodoItem(<string> tabSubmit.data.action),
+                    <number> tabSubmit.data.taskId);
+                break;
+            }
         }
-        case "learn": {
-          this.likeCountObj.likeCount = 0;
-          const card = AdaptiveCards.declare<DataInterface>(rawLearnCard).render(this.likeCountObj);
-          await context.sendActivity({ attachments: [CardFactory.adaptiveCard(card)] });
-          break;
+
+        return tabResp;
+    }
+
+    // Message extension Code
+    async handleTeamsMessagingExtensionFetchTask(context: TurnContext, action: MessagingExtensionAction): Promise<MessagingExtensionActionResponse> {
+        console.log("Trying To Fetch Task Module From Messaging Extension");
+        const MEActionResp: MessagingExtensionActionResponse = {
+            task: {
+                type: "continue",
+                value: {},
+            },
+        };
+        
+        let taskInfo: TaskModuleTaskInfo;
+        switch (action.commandId) {
+            case "createTodo": {
+                taskInfo = createNewItemTaskInfo(action.messagePayload.body.content);
+                break;
+            }
         }
-        /**
-         * case "yourCommand": {
-         *   await context.sendActivity(`Add your response here!`);
-         *   break;
-         * }
-         */
-      }
-
-      // By calling next() you ensure that the next BotHandler is run.
-      await next();
-    });
-
-    this.onMembersAdded(async (context, next) => {
-      const membersAdded = context.activity.membersAdded;
-      for (let cnt = 0; cnt < membersAdded.length; cnt++) {
-        if (membersAdded[cnt].id) {
-          const card = AdaptiveCards.declareWithoutData(rawWelcomeCard).render();
-          await context.sendActivity({ attachments: [CardFactory.adaptiveCard(card)] });
-          break;
+
+        MEActionResp.task.value = taskInfo;
+        
+        return MEActionResp;
+    }
+
+    // Handle submits from Messaging Extension.
+    async handleTeamsMessagingExtensionSubmitAction(context: TurnContext, action: MessagingExtensionAction): Promise<MessagingExtensionActionResponse> {
+        console.log("Trying To Submit Task Module From Messaging Extension");
+        const MEActionResp: MessagingExtensionActionResponse = {
+            task: {
+                type: "continue",
+                value: {},
+            },
+        };
+
+        const user: User = {
+            aadObjectId: context.activity.from.aadObjectId,
+            userName: context.activity.from.name
+        };
+
+        let taskInfo: TaskModuleTaskInfo;
+        switch (action.commandId) {
+            case "createTodo": {
+                var content: string;
+                try {
+                    await handleNewItemAction(user.aadObjectId, action.data);
```

Comment:
may call the function createToDoItem. By the way, instead ot pass action.data, we can pass actual values (dueDate, content) to the function. So the code may be `await createToDoItem(user.aadObjectId, action.data.dueDate, action.data.content)`